tests/urls.js

```diff
@@ -0,0 +1,51 @@
+const fs = require('fs').promises;
+const core = require('@actions/core');
+const AbortController = require('abort-controller');
+
+// Helper function to create a timeout promise
+function timeout(ms) {
+  return new Promise((_, reject) => setTimeout(() =>
+    reject(new Error('timeout')), ms));
+}
+
+async function checkURL(url, file) {
+  const controller = new AbortController();
+  const signal = controller.signal;
+
+  try {
+    const res = await Promise.race([
+      fetch(url, {
+        headers: {
+          'User-Agent': '2factorauth/URLValidator (+https://2fa.directory/bots)',
+        },
+        signal,
+      }),
+      timeout(2000).then(() => controller.abort()),
+    ]);
+
+    if (res.ok)
+      return true;
+    else if (res.status !== 403)
```

Comment:
Ignoring status code 403 because we most often encounter that because of WAF rules meant to block bots and web scrapers.
![http_403_ok](https://github.com/user-attachments/assets/73442b77-a25b-4829-9a00-a2a6762eba26)

Reply:
Bad programmers: the bane of everyone's existence.